evm/basin_storage/src/BasinStorage.sol

```diff
@@ -75,194 +64,74 @@ contract BasinStorage is AccessControl {
         emit PubCreated(pub, owner);
     }
 
-    /// @dev Adds the deal info, given deals and a pub.
+    /// @dev Adds the CID for the pub and timestamp.
     ///      Can only be called by the Pub Admin.
-    /// @param deals The Filecoin deal object.
-    /// @param pub The publication to add the deal info for.
-    function addDeals(
+    /// @param pub The publication to add the CID for.
+    /// @param cid The content id for object in the Filecoin deal.
+    /// @param timestamp The timestamp provided by data owner.
+    function addCID(
         string calldata pub,
-        DealInfo[] calldata deals
+        string calldata cid,
+        uint256 timestamp
     ) external onlyRole(PUB_ADMIN_ROLE) {
-        uint256 epoch = block.number;
         address owner = _pubs[pub];
-
         // Pub must already exist
         if (owner == address(0)) {
             revert PubDoesNotExist(pub);
         }
-
-        // Loop through the input deals and add them against an epoch
-        for (uint256 i = 0; i < deals.length; i++) {
-            _deals[pub][epoch].push(deals[i]);
-            _pubDealCount[pub]++;
-            emit DealAdded(deals[i].id, pub, owner, deals[i].cid);
-        }
+        _cids[pub][timestamp].push(cid);
+        _pubCIDCount[pub]++;
```

Comment:
total number cids for a pub is maintained in storage for iterating during retrieval